climada/engine/impact.py

```diff
@@ -1372,7 +1372,147 @@ def _selected_exposures_idx(self, coord_exp):
+          impact.date = np.array(dates)
+          
+          
+          #check if exposure is consistent
+          isin=[np.isin(imp.coord_exp,imp_list[0].coord_exp) for imp in imp_list[0::]]
+          tot_vals=set([imp.tot_value for imp in imp_list])
+          if (not np.all(isin)) or (len(tot_vals)>1):
+               raise ValueError("Impacts are not based on the same exposure. Please make sure exposures are identical."
+                                "The impacts are incompatible and cannot be concatenated.")
+          exp_descriptions= {imp.tag['exp'].description for imp in imp_list if imp.tag['exp'].description != ''}
+          if len(exp_descriptions) > 1:
+               raise ValueError(f"The given impacts are based on exposures with different descriptions: {exp_descriptions}. "
+                            "The impacts are incompatible and cannot be concatenated.")      
+          impact.coord_exp=imp_list[0].coord_exp
+          impact.tot_value=imp_list[0].tot_value
+          
+          #fill remaining attributes
+          impact.tag=imp_list[0].tag
+
+          #event IDs
+          event_ids=[event_id for imp in imp_list for event_id in imp.event_id]
+          if len(event_ids)!=len(set(event_ids)):
+              raise ValueError("Duplicate event IDs found.")
+          impact.event_id=np.array(event_ids)
+          
+          #event names
+          event_names=[event_name for imp in imp_list for event_name in imp.event_name]
+          if len(event_names)!=len(set(event_names)):
+              raise ValueError("Duplicate event names found.")   
+          impact.event_name = np.array(event_names,dtype=object)
+
+          #impact matrix
+          imp_mats=[imp.imp_mat for imp in imp_list]
+          impact.imp_mat=sparse.vstack(imp_mats)
+          
+          #concatenate remaining attributes
+          at_events=[imp.at_event for imp in imp_list]
+          impact.at_event = np.concatenate(at_events,axis=0)
+          eai_exps=[imp.eai_exp for imp in imp_list]
+          impact.eai_exp = np.nansum(eai_exps,axis=0) 
+          aai_aggs=[imp.aai_agg for imp in imp_list]
+          impact.aai_agg = np.nansum(aai_aggs)
+
+        elif concat_type=='exposure':
+            
+          ## some possibly useful code snippets for this option
+          ##merge coordinates (only if exposure is not the same)
+          ##check if there are any additional exposure coordinates in the new impact
+          #compare_coords=np.isin(impact.coord_exp,self.coord_exp)
+          #additional_coords=[impact.coords[ind] for ind, x in enumerate(compare_coords) if not np.all(x)]
+          #imp.coord_exp=np.concatenate(self.coord_exp,additional_coords)
+        
+          raise ValueError("Concatenation of impacts with different exposures not yet implemented."
+                             "Feel free to do so!")
+     
+
+          
+
+
+
+        return impact
```

Comment:
```suggestion
                return cls(
            event_id = event_id,
            event_name = event_name,
            date = date,
            frequency = frequency,
            coord_exp = coords,
            crs = crs,
            unit = value_unit,
            tot_value = total_value,
            eai_exp = eai_exp,
            at_event = at_event,
            aai_agg = aai_agg,
            imp_mat = imp_mat if imp_mat is not None else sparse.csr_matrix((0, 0)),
            tag = tag
      )

```
Something along these lines, you will have to update the variables.